### DISCUSSION TITLE
Support renovating Terrafrom Modules from private Git repositories using shallow clone URL argument

### DISCUSSION THREAD

**Discussion by Moglum** [renovatebot/renovate#21266] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Hi folks,
Renovate already supports the following format of `source` URLs for Terraform modules:
```
git::ssh://git@git.example.com/modules/foo-module.git//hub?ref=v7.1.0
```

Based on [Terraform documentation](https://developer.hashicorp.com/terraform/language/modules/sources#shallow-clone), users can also specify an additional `depth` argument in the URL
```
git::ssh://git@git.example.com/modules/foo-module.git//hub?depth=1&ref=v7.1.0
```

The [currently used Regex](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/terraform/extractors/others/modules.ts#L18) doesn't support the argument - https://regex101.com/r/RASnRo/1

That is blocking our usage of Renovate at the moment, as we leverage shallow clone to save time during `terraform init` 

I propose slightly modifying the regex to allow parsing the URLs even with the `depth` argument specified https://regex101.com/r/RASnRo/2

```
(?:git::)?(?<url>(?:(?:http|https|ssh):\/\/)?(?:.*@)?(?<path>.*\/(?<project>.*\/.*)))\?.*ref=(?<tag>.*)$
```
instead of 
```
(?:git::)?(?<url>(?:(?:http|https|ssh):\/\/)?(?:.*@)?(?<path>.*\/(?<project>.*\/.*)))\?ref=(?<tag>.*)$  
```

I'm ready to submit a PR with that change if you consider it a good idea. 
Many thanks for considering my request.

**Comment by secustor** [comment]
Feel free to open a PR.

**Comment by Moglum** [comment]
PR created - https://github.com/renovatebot/renovate/pull/21287

**Comment by Moglum** [comment]
Resolved in https://github.com/renovatebot/renovate/pull/21287
